### PR TITLE
Fork RNTester script files and update 'React' to 'react'

### PR DESCRIPTION
**Pull request by ddalp**
This change addresses some issues found while attempting to upgrade to RN61 and remove our dependency on Microsoft's fork of React Native. Would like to check in those changes to current master so the future upgrade change list can be less complicated.

1. We are currently getting RNTester from the Microsoft's fork of React-Native, but it is not published with the React-Native NPM package. So we will need to fork it somewhere in VNext after we migrate to RN 61 and Microsoft RN fork is removed. Ideally Facebook should publish a separate package for RNTester but we have to fork it for now. In this change, we are forking the latest RNTester's JS directory to vNext\RNTesterFork, and build scripts and metro config files are also updated to use or exclude the RNTesterFork directory.

2. After we upgrade RN.61, it requires case sensitive. We have some forked JS files needs to be updated from require "React" to require "react".

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3944)

### CODE REVIEW

**Comment by acoates-ms** [comment]
This is just a flat copy/paste of RNTester?  Is this 0.60 or 0.61 version of RNTester?

**Comment by NickGerleman** [comment]
nit: Should we leave "Fork" out of the directory name? I'm wondering if the name will give people the idea it's okay to introduce changes into it.

**Comment by ddalp** [comment]
I think I should use the RNTester from Microsoft Forked RN depot for now. We can then update it to .61 RNTester when we upgrade RNW to use RN .61

---
In reply to: [577923565](https://github.com/microsoft/react-native-windows/pull/3944#issuecomment-577923565) [](ancestors = 577923565)

**Comment by ddalp** [comment]
Maybe RNTesterCopy?

---
In reply to: [577932265](https://github.com/microsoft/react-native-windows/pull/3944#issuecomment-577932265) [](ancestors = 577932265)

File: vnext/.npmignore [hunk 1]
```diff
@@ -23,6 +23,7 @@ ReactWindows-Universal.sln
 ReactWindows-Desktop.sln
 RNTester.*
 RNTester/
+RNTesterFork/
```
Comment (NickGerleman) [inline vnext/.npmignore]: Needs to be updated

File: packages/playground/metro.config.js [hunk 2]
```diff
@@ -67,7 +72,6 @@ module.exports = {
           .replace(/[/\\\\]/g, '[/\\\\]')}.*`,
       ),
     ]),
-    hasteImplModulePath: path.resolve(__dirname, 'hasteImpl.js'),
```
Comment (kmelmon) [inline packages/playground/metro.config.js]: >hasteImplModulePath: path.resolve(__dirname, 'hasteImpl.js'), [](start = 4, length = 61)

Did you mean to remove this?  Looks like something for 61.

**Comment by kmelmon** [comment]
I agree with that plan - from a quick diff it looks like the JS in facebook repo is the same between 60 and 61 but let's play it safe for now.  I've created a TODO in the project board to deal with this when we go to 61.

---
In reply to: [577933815](https://github.com/microsoft/react-native-windows/pull/3944#issuecomment-577933815) [](ancestors = 577933815,577923565)

**Comment by NickGerleman** [comment]
@didalp I realized that there's a small issue with the rn copy logic right now. Copying RNTesterCopy from the base path will only work for the vnext directory, snd not packages like react-native-win32

**Comment by NickGerleman** [comment]
I'm not entirely sure the blacklist works on stock React Native. I get a bunch of haste related errors merging the change into the branch I've been using to test Facebook 0.60.

Something that could maybe sidestep the whole problem is removing the JS copy and just naming this new directory RNTester. We could also get less disk space usage when doing that.

https://github.com/microsoft/react-native-windows/pull/3945
image

**Comment by ddalp** [comment]
Yeah, it failed and I will try to fix it.

---
In reply to: [577966058](https://github.com/microsoft/react-native-windows/pull/3944#issuecomment-577966058) [](ancestors = 577966058)

**Comment by ddalp** [comment]
thanks.

---
In reply to: [577949246](https://github.com/microsoft/react-native-windows/pull/3944#issuecomment-577949246) [](ancestors = 577949246,577933815,577923565)

File: packages/playground/metro.config.js (hunk 2 above)
Comment (ddalp) [reply packages/playground/metro.config.js]: It is accident, but I kind of feel this is not needed anymore event for 60. But I will put it back and create a issue on our 61 project board to visit all hasteImpl.js.

---
In reply to: [370431313](https://github.com/microsoft/react-native-windows/pull/3944#discussion_r370431313) [](ancestors = 370431313)

**Pull request merged**